utils/audio.js

```diff
@@ -1,18 +1,101 @@
 'use strict';
 
-module.exports = {
-  buildSfxCollection(game, componentName, clipCount) {
-    // FIXME: Make this a class.
+class Audio {
+  constructor(game) {
+    this.game = game;
+  }
+
+  buildSfxCollection(componentName, clipCount) {
     const sounds = [];
 
     for (let i = 0; i < clipCount; i++) {
-      sounds.push(this.addSfx(game, componentName, i));
+      sounds.push(this.addSfx(componentName, i));
     }
 
     return sounds;
-  },
+  }
+
+  addSfx(componentName, clipId) {
+    return this.game.add.audio(`${componentName}-${clipId}-sfx`);
+  }
+
+  // Play sequence of non-looping sounds
+  playSequence(soundKeys) {
+    const sounds = soundKeys.map(key => this.game.add.audio(key));
+
+    this.playSoundsInOrder(sounds);
+  }
+
+  playSoundsInOrder(sounds) {
+    if (sounds.length > 0) {
+      sounds[0].play();
+      sounds[0].onStop.addOnce(() => {
+        this.playSoundsInOrder(sounds.slice(1));
+      });
+    }
+  }
+
+  playSequenceWithCrossfades(soundKeys, dividingFactor) {
+    if (!dividingFactor || dividingFactor < 1) {
```

Comment:
Oh, and not sure if you understood what I was referring to, but I was suggesting that `dividingFactor = dividingFactor || 10` could replace the entire conditional block, if you expect the input to be integer and non-negative. If you're trying to avoid non-integer values causing the fade time to be greater than the duration, then you'd still need the `dividingFactor < 1`, though you could do it as a ternary.

I leave it up to you :).